### PR TITLE
OS X -> macOS

### PR DESCRIPTION
**Pull request by z11h**

### CODE REVIEW

**Comment by laughedelic** [comment]
Why not to do the same then in `docs/building/osx.md` and the links to it?

**Comment by andschwa** [comment]
From Wikipedia:

> Major revision 10, from 2001 to present, was originally branded "Mac OS X", renamed to "OS X" in 2012,[4] and announced to be renamed to "macOS" in the upcoming version in 2016.

We've gone with OS X as that was the correct name last year. If Apple has now officially rebranded to macOS, we should update the usage everywhere throughout the documentation.

**Comment by ZakariaRidouh** [comment]
Yes, that was the plan. I can update all the documentation to macOS if you're interested.


-- Zakaria Ridouh






On Fri, Aug 19, 2016 at 9:45 AM -0700, "Andy Schwartzmeyer" <notifications@github.com> wrote:





From Wikipedia:

> Major revision 10, from 2001 to present, was originally branded "Mac OS X", renamed to "OS X" in 2012,[4] and announced to be renamed to "macOS" in the upcoming version in 2016.

We've gone with OS X as that was the correct name last year. If Apple has now officially rebranded to macOS, we should update the usage everywhere throughout the documentation.

**Comment by andschwa** [comment]
That'd be great :smile:

**Comment by ZakariaRidouh** [comment]
@andschwa fixed 😄

**Comment by TravisEz13** [comment]
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )

**Comment by ZakariaRidouh** [comment]
The conflicts have been fixed. @TravisEz13

File: README.md [hunk 1]
```diff
@@ -27,22 +27,22 @@ You can download and install a PowerShell package for any of the following platf
...
-[rl-osx]: https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.9/powershell-6.0.0-alpha.9.pkg
+[rl-macOS]: https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.9/powershell-6.0.0-alpha.9.pkg
 
 [installation]: docs/installation
 [in-windows]: docs/installation/windows.md#msi
 [in-ubuntu14]: docs/installation/linux.md#ubuntu-1404
 [in-ubuntu16]: docs/installation/linux.md#ubuntu-1604
 [in-centos]: docs/installation/linux.md#centos-7
-[in-osx]: docs/installation/linux.md#os-x-1011
+[in-osx]: docs/installation/linux.md#macos-1011
```
Comment (alexandair) [inline README.md]: Please, change [in-osx] to [in-macos]

File: docs/dev-process/coding-guidelines.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ There are 3 primary preprocessor macros we define during builds:
 
 * DEBUG - guard code that should not be included in release builds
 * CORECLR - guard code that differs between Full CLR and CoreCLR
-* UNIX - guard code that is specific to Unix (Linux and Mac OS X)
+* UNIX - guard code that is specific to Unix (Linux and Mac macOS)
```
Comment (alexandair) [inline docs/dev-process/coding-guidelines.md]: I think (Linux and macOS) is enough. We don't need "Mac macOS".

**Comment by ZakariaRidouh** [comment]
@alexandair I've fixed the corresponding mistakes 😄

**Comment by lzybkr** [comment]
We also have the variable `$IsOSX` and the api `[System.Management.Automation.Platform]::IsOSX` to consider.

Of course, once we land on the right api, we'll need to stick with it for compat, we don't want two apis just because marketing changed their minds.

**Comment by andschwa** [comment]
@lzybkr that's an excellent point, but I would recommend sticking with `OSX` in code to be consistent with [.NET Core's established convention](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs#L15) (which is all that `$IsOSX` maps to).

**Comment by ZakariaRidouh** [comment]
@andschwa I agree. The `OSX` naming scheme would have to be a change in the `.Net Core` project. For now, I don't think there is anything we can do about that 😄

**Comment by ZakariaRidouh** [comment]
Is this ready to merge? @andschwa

**Comment by andschwa** [comment]
I'm good with this, merging.

**Pull request merged**